### PR TITLE
Configure SQL TLS environment variables in server-job

### PR DESCRIPTION
**Pull request by grzegorz8**
## What was changed

Added support for `SQL_TLS*` properties in schema jobs (`server-job.yaml`).

## Why?

Currently, when PostgreSQL database is secured with TLS, schema jobs are not properly configured.

In the Pull Request we reuse `server.config.persistence.*.sql.tls`, `server.config.additionalVolumes` and `server.config.additionalVolumeMounts` properties..

## Checklist

1. Closes 
No related issues.

2. How was this tested:

3. Any docs updates needed?

No updates needed.

### CODE REVIEW

**Comment by LukaGiorgadze** [comment]
@grzegorz8
May I ask you - what does it mean "schema jobs are not properly configured"?

**Comment by grzegorz8** [comment]
> @grzegorz8 May I ask you - what does it mean "schema jobs are not properly configured"?

I meant that TLS related env variables are not set (SQL_TLS_KEY_FILE, SQL_TLS_CERT_FILE, etc).

**Comment by arkoc** [comment]
`SQL_TLS=true` is required option for conencting to Azure Postgres Flexible Servers. Probably it is required for other 'serverless' managed databases, but we only tested on Azure.

**Comment by debugger24** [comment]
Hi @grzegorz8 any update on this PR ?

**Review by debugger24 — APPROVED** [review]
LGTM

**Review by MasonLegere — APPROVED** [review]
This looks good to me - tested with Postgres 15 on RDS

**Comment by syugoman-armenotech** [comment]
Hi everyone! Any updates for this PR?

**Comment by syugoman-armenotech** [comment]
@grzegorz8 Hi! Can you please update an example? It will be helpful 
https://github.com/temporalio/helm-charts/blob/master/charts/temporal/values/values.postgresql.yaml

**Comment by grzegorz8** [comment]
> @grzegorz8 Hi! Can you please update an example? It will be helpful https://github.com/temporalio/helm-charts/blob/master/charts/temporal/values/values.postgresql.yaml

Example updated. Please check if it looks fine.

**Comment by devinargenta** [comment]
Hello! Wanted to bump this as it would be useful for us. Thank you!

**Comment by moss2k13** [comment]
can someone clarify the status of this PR?

we are using these changes without issues with postgresql versions 15.x

File: charts/temporal/values/values.postgresql.yaml [hunk 1]
```diff
@@ -40,6 +43,17 @@ server:
           #  enabled: true
           #  enableHostVerification: true
           #  serverName: _HOST_ # this is strictly required when using serverless CRDB offerings
+          #  caFile: /path/to/certs/<CA-file> # Here we assumed that caFile, certFile, keyFile are stored in one secret mounted as 'secret-with-certs' (see: server.additionalVolumes and server.additionalVolumeMounts sections).
+          #  certFile: /path/to/certs/<client-cert-file>
+          #  keyFile: /path/to/certs/<client-key-file>
+
+# additionalVolumes:
```
Comment (robholland) [inline charts/temporal/values/values.postgresql.yaml:50]: These should be indented so that they are under `server`.

**Review by robholland — CHANGES_REQUESTED** [review]

**Comment by robholland** [comment]
Looks good, one tiny tweak.

File: charts/temporal/values/values.postgresql.yaml (hunk 1 above)
Comment (grzegorz8) [reply charts/temporal/values/values.postgresql.yaml]: @robholland Thanks, fixed.

**Review by robholland — APPROVED** [review]

**Comment by robholland** [comment]
Thanks for your contribution :) For future reference, when you've attended to feedback in PR, if you can re-request review from the reviewer, it helps us spot when something is ready for us to look at again.

**Comment by esn89** [comment]
Does this additionalVolumeMount nest in:
```server.config.additionalVolumes```

Like what the original post says?

Or nested under `server.additionalVolumes` as per:
https://github.com/temporalio/helm-charts/pull/411/files#diff-5f8eb04d49dd7caffd33da659dd7bae84435c114a6255319b2965dcf6a2536abR254

**Comment by grzegorz8** [comment]
> Does this additionalVolumeMount nest in: `server.config.additionalVolumes`
> 
> Like what the original post says?
> 
> Or nested under `server.additionalVolumes` as per: https://github.com/temporalio/helm-charts/pull/411/files#diff-5f8eb04d49dd7caffd33da659dd7bae84435c114a6255319b2965dcf6a2536abR254

The description is wrong, the code is right. Sorry for confusion.

**Pull request merged**